soc/riscv/riscv-ite/it8xxx2/soc.c

```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright (c) 2020 ITE Corporation. All Rights Reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+#include <kernel.h>
+
+#include <device.h>
+#include <init.h>
+#include <sys/util.h>
```

Comment:
why do we need these headers included?

Reply:
> DTS:
> 
> * Seems like the dts/riscv/it8xxx2_evb.dtsi  should be dts/riscv/it8xxx2.dtsi for the SoC (as I assume EVB means eval board).
> * Its general convention for address to be lowercase (ie `interrupt-controller@F03F00` should be `interrupt-controller@f03f00`
> * looking at the datasheet the INTC supports edge/level and polarity, so those should be encoded in the devicetree.
> 
> Recommend introducing include/dt-bindings/irq.h that has the following defines:
> 
> ```
> #define IRQ_TYPE_NONE           0
> #define IRQ_TYPE_EDGE_RISING    1
> #define IRQ_TYPE_EDGE_FALLING   2
> #define IRQ_TYPE_EDGE_BOTH      (IRQ_TYPE_EDGE_FALLING | IRQ_TYPE_EDGE_RISING)
> #define IRQ_TYPE_LEVEL_HIGH     4
> #define IRQ_TYPE_LEVEL_LOW      8
> ```

Excuese, I try to add this, but there are some issues as follow:
a. I can't find the include/dt-bindings/irq.h file, so I create a file at include/dt-bindings/interrupt-controller/ite-it8xxx2.h. (If this is not appropriate, please let me know.)  
b. I thought the setting intc trigger mode should be coding in arch_irq_priority_set(irq_p, priority_p, flags_p) under riscv/soc, but the function defindef in include/arch/riscv/arch.h is only two parameter. the z_irq_priority_set(irq_p, priority_p, flags_p) will pass three parameters too. Did I make a mistake? Should I coding this at another place?
Thanks you.

Reply:
> Excuese, I try to add this, but there are some issues as follow:
> a. I can't find the include/dt-bindings/irq.h file, so I create a file at include/dt-bindings/interrupt-controller/ite-it8xxx2.h. (If this is not appropriate, please let me know.)

Might not have been clear, we need to create `include/dt-bindings/irq.h`

> b. I thought the setting intc trigger mode should be coding in arch_irq_priority_set(irq_p, priority_p, flags_p) under riscv/soc, but the function defindef in include/arch/riscv/arch.h is only two parameter. the z_irq_priority_set(irq_p, priority_p, flags_p) will pass three parameters too. Did I make a mistake? Should I coding this at another place?

trigger mode would be handled via flags paramater.

Reply:
Hi,
@galak, @mnkp 
Excuese me for bothering you,  please review again, thank you.